src/lib/components/table/QTable.svelte

```diff
@@ -2,45 +2,45 @@
   import { QIcon, QSelect, QBtn } from "$lib";
   import type { QTableProps, QTableColumn, QTableRow, QTableSort } from "./props";
 
-  export let columns: QTableProps["columns"] = [],
-    rows: QTableProps["rows"] = [],
-    flat: QTableProps["flat"] = undefined,
-    bordered: QTableProps["bordered"] = undefined,
-    dense: QTableProps["dense"] = false,
-    userClasses: QTableProps["userClasses"] = "";
-  export { userClasses as class };
+  let {
+    columns = [],
+    rows = [],
+    flat = undefined,
+    bordered = undefined,
+    dense = false,
+    bodyCell = undefined,
+    ...props
+  }: QTableProps = $props();
 
   function getField(fieldRaw: QTableColumn["field"], row: QTableRow): string | number {
     return typeof fieldRaw === "function" ? fieldRaw(row) : row[fieldRaw];
   }
 
-  let page = 1;
-  let rowsPerPage = 5;
-  let rowsPerPageOptions = [5, 10, 25, 50].map((e) => ({
-    label: e.toString(),
-    value: e.toString(),
-  }));
-
-  let sort: QTableSort = null;
-  let rowsSorted: QTableRow[] = rows;
-  let rowsPaginated: QTableRow[] = [];
-  let numberFrom: number = 1;
-  let numberTo: number = 5;
-  let numberOf: number = rows.length;
-
-  $: numberFrom = rowsPerPage * page - rowsPerPage + 1;
-  $: numberTo = rowsPerPage * page > rows.length ? rows.length : rowsPerPage * page;
-  $: numberOf = rows.length;
-  $: rowsPaginated = rowsSorted.slice(numberFrom - 1, numberTo);
-  $: {
+  let page = $state(1);
+  let rowsPerPage = $state(5);
+  let rowsPerPageOptions = $state(
+    [5, 10, 25, 50].map((e) => ({
+      label: e.toString(),
+      value: e.toString(),
+    }))
+  );
+  let sort: QTableSort = $state(null);
+
+  const numberFrom: number = $derived(rowsPerPage * page - rowsPerPage + 1);
+  const numberTo: number = $derived(
+    rowsPerPage * page > rows.length ? rows.length : rowsPerPage * page
+  );
+  const numberOf: number = $derived(rows.length);
```

Comment:
Do you really need to write the type explicitly for those 3?

Reply:
No, but now it's sure the expressions actually return the correct type - at least from a TS perspective.